packages/expo-updates/ios/EXUpdates/AppLoader/EXUpdatesFileDownloader.m

```diff
@@ -391,14 +402,90 @@ - (void)downloadManifestFromURL:(NSURL *)url
   } errorBlock:errorBlock];
 }
 
-- (void)_createUpdateWithManifest:(NSMutableDictionary *)mutableManifest
+- (nullable EXUpdatesCodeSigningConfiguration *)codeSigningConfiguration {
+  if (!_config.codeSigningCertificate) {
+    return nil;
+  }
+
+  NSError *error;
+  EXUpdatesCodeSigningConfiguration *codeSigningConfiguration = [[EXUpdatesCodeSigningConfiguration alloc] initWithCertificate:_config.codeSigningCertificate
+                                                                                                                      metadata:_config.codeSigningMetadata
+                                                                                                                         error:&error];
+  if (error) {
+    NSString *message;
+    if (error.code == EXUpdatesCodeSigningConfigurationErrorCertificateParseError) {
+      message = @"Could not parse code signing certificate";
+    } else if (error.code == EXUpdatesCodeSigningConfigurationErrorCertificateValidityError) {
+      message = @"Certificate not valid";
+    } else if (error.code == EXUpdatesCodeSigningConfigurationErrorCertificateDigitalSignatureNotPresentError) {
+      message = @"Certificate digital signature not present";
+    } else if (error.code == EXUpdatesCodeSigningConfigurationErrorCertificateMissingCodeSigningError) {
+      message = @"Certificate missing code signing extended key usage";
+    }
+    
+    @throw [NSException exceptionWithName:NSInternalInconsistencyException reason:message userInfo:nil];
+  }
+  
+  return codeSigningConfiguration;
+}
+
+- (void)_createUpdateWithManifest:(NSDictionary *)manifest
+                 manifestBodyData:(NSData *)manifestBodyData
                   manifestHeaders:(EXUpdatesManifestHeaders *)manifestHeaders
                        extensions:(NSDictionary *)extensions
                          database:(EXUpdatesDatabase *)database
                        isVerified:(BOOL)isVerified
                      successBlock:(EXUpdatesFileDownloaderManifestSuccessBlock)successBlock
                        errorBlock:(EXUpdatesFileDownloaderErrorBlock)errorBlock
 {
+  @try {
```

Comment:
NSExceptions and try/catch in obj-c are generally reserved for programming errors, whereas in the rest of the module we've used NSError patterns for errors like this that are expected to happen at runtime with certain inputs. Would prefer to stick to that here too, unless there's a good reason that I'm missing for using try/catch/NSException.